rust/arrow/src/ffi.rs

```diff
@@ -395,6 +397,12 @@ unsafe extern "C" fn release_array(array: *mut FFI_ArrowArray) {
     // take ownership of `private_data`, therefore dropping it
     Box::from_raw(array.private_data as *mut PrivateData);
 
+    // release children
+    for i in 0..array.n_children {
+        let child = *array.children.add(i as usize);
+        release_array(child);
```

Comment:
You should call the release callback stored on `child`, not `release_array` directly.

Reply:
Note you must do the same thing in `release_schema` as well.